### PR TITLE
make deploy target adapter for further support webapp deployment slot

### PR DESCRIPTION
**Pull request by SummerSun**

### CODE REVIEW

**Comment by brunoborges** [comment]
Hi @SummerSun... I'm not sure what is the purpose of this PR and what feature it provides. Can you please update the description? 

While I understand the goal (adapter for webapp), not sure it actually delivers that more than just an abstraction of the Maven Plugin API, which in that case it makes little sense because this is a Maven plugin after all.

**Comment by SummerSun** [comment]
@brunoborges 

I am fixing the issue #151 which is to support deployment slot deploy.
This is the part1 change list for supporting deployment slot.

The adapter enable us could pass in any deploy target implments .warDeploy API, such as DeploymentSlot (will implements the adapter interface) , then we don't need to update handlers to achieve deployment slot deploy.

**Comment by brunoborges** [comment]
Got it, thanks. Would be helpful to have that sort of detail in future PRs. Or at least a link to the issue a PR intends to fix/implement.

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/deployadapter/IDeployTargetAdapter.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ * license information.
+ */
+
+package com.microsoft.azure.maven.webapp.deployadapter;
+
+import com.microsoft.azure.management.appservice.PublishingProfile;
+
+import java.io.File;
+
+public interface IDeployTargetAdapter {
```
[the remark this answers is not in the document]
Comment (SummerSun) [reply azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/deployadapter/IDeployTargetAdapter.java]: We have two kinds of deploy target after adding support for deployment slot.

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/deployadapter/IDeployTargetAdapter.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    void warDeploy(File war, String contextPath);
```
[the remark this answers is not in the document]
Comment (SummerSun) [reply azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/deployadapter/IDeployTargetAdapter.java]: Both deployment slot and web app define the API warDeploy in its independent file, we create this adapter to make sure we have one interface for all targets that implement warDeploy to do the publish.

**Pull request merged**